src/main/java/com/khutircraftubackend/config/MailConfig.java

```diff
@@ -0,0 +1,56 @@
+package com.khutircraftubackend.config;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.JavaMailSenderImpl;
+
+import java.util.Properties;
+
+/**
+ * Клас MailConfig налаштовує параметри для відправки електронної пошти за допомогою JavaMailSender.
+ * <p>
+ * Цей клас використовує Spring Framework для створення конфігурації, що включає налаштування
+ * SMTP-сервера для відправки електронної пошти.
+ * </p>
+ */
+
+@Configuration
+public class MailConfig {
+
+    @Value("${spring.mail.host}")
+    private String host;
+
+    @Value("${spring.mail.port}")
+    private int port;
+
+    @Value("${spring.mail.username}")
+    private String username;
+
+    @Value("${spring.mail.password}")
+    private String password;
+
+    @Value("${spring.mail.properties.mail.smtp.auth}")
+    private String smtpAuth;
+
+    @Value("${spring.mail.properties.mail.smtp.starttls.enable}")
+    private String starttlsEnable;
```

Comment:
if `spring.mail.properties.mail.smtp.auth` and `spring.mail.properties.mail.smtp.starttls.enable` could be flatten we could consider using @ConfigurationProperties

```yml
  mail:
    host: ${MAIL_HOST}
    port: ${MAIL_PORT}
    username: ${MAIL_USERNAME}
    password: ${MAIL_PASSWORD}
    smtp:
       auth: true
       starttls:
         enable: true
```
or in general, 
do you think it would be reasonable to reorganise this block in yaml file as shown above?

Reply:
Do you mean to remove the MailConfig class and use SpringBoot auto-created JavaMailSender?

Reply:
nope :) 
check this example out: [mkyong.com - ConfigurationProperties](https://mkyong.com/spring-boot/spring-boot-configurationproperties-example/#configurationproperties)